modules/pam_motd/pam_motd.c

```diff
@@ -97,9 +98,180 @@ static void try_to_display_directory(pam_handle_t *pamh, const char *dirname)
     }
 }
 
+/*
+ * Get the number of occurrences of substring DELIM in ARG.
+ */
+static int get_num_occurrences(const char *arg, const char *delim)
+{
+    if (arg == NULL || delim == NULL) {
+	return 0;
+    }
+
+    int num_occurrences = 0;
+    const char *arg_ptr = arg;
+    while ((arg_ptr = strstr(arg_ptr, delim))) {
+	num_occurrences++;
+	arg_ptr += strlen(delim);
+    }
+    return num_occurrences;
+}
+
+/*
+ * Split a string ARG into an array of NUM_STRS strings or until strtok
+ * returns NULL, whichever is lower. Returns newly allocated array of
+ * strings ARG_SPLIT.
+ */
+static char ** split_string(char *arg, int num_strs, const char *delim)
+{
+    if (arg == NULL || delim == NULL) {
+	return NULL;
+    }
+    if (num_strs < 1) {
+	return NULL;
+    }
+
+    char **arg_split = NULL;
+    char *arg_extracted = NULL;
+    int i = 0;
+
+    arg_split = (char **)malloc(sizeof(char *) * num_strs);
+    if (!arg_split) {
```

Comment:
Here and everywhere else, I think it would be better to have null-pointer checks being explicit (i.e. `!= NULL`).

Reply:
Agreed, making them explicit